File: src/Systems/Swift/patterns/ActiveObject.swift
enum ActiveObjectExample {
    static func run() -> Bool {
        var value = 0
        let queue: [() -> Void] = [{ value += 3 }, { value *= 4 }]
        let before = value
        queue.forEach { $0() }
        return before == 0 && value == 12
    }
}

File: src/Systems/Fortran/patterns/active_object.f90
module active_object_example
  implicit none
contains
  logical function run()
    integer :: value, queue(2), i
    value = 0; queue = [3, 4]
    do i = 1, size(queue)
      if (i == 1) value = value + queue(i)
      if (i == 2) value = value * queue(i)
    end do
    run = value == 12
  end function
end module
